core/xml.js

```diff
@@ -128,9 +134,20 @@ Blockly.Xml.allFieldsToDom_ = function(block, element) {
  * Encode a block subtree as XML.
  * @param {!Blockly.Block} block The root block to encode.
  * @param {boolean=} opt_noId True if the encoder should skip the block ID.
- * @return {!Element} Tree of XML elements.
+ * @return {Element} Tree of XML elements.
  */
 Blockly.Xml.blockToDom = function(block, opt_noId) {
+  // Skip over insertion markers.
+  if (block.isInsertionMarker()) {
+    var child = block.getChildren(false)[0];
+    if (child) {
+      return Blockly.Xml.blockToDom(child);
+    } else {
+      return null;
```

Comment:
If i'm understanding what you're saying correctly, this ``null`` would never actually happen?
If so, can we just cast child to non-nullable and maintain that blockToDom returns ``!Element``

Reply:
So this function /will/ return null if it is passed an insertion marker.

But it will never return null from the "create" event, because the create event always passes it a real block.

-----

The problem this is solving is making sure insertion markers don't get added to xml if the developer happens to serialize the workspace while the user is dragging. But maybe there's a better way to handle this than returning null (since it seems to really mess up typing).

I'm going to split this off into a second PR b/c I don't want it to hold up fixing the code gen. Maybe we can talk about it more over on #2322